sgx-types/src/secs.rs

```diff
@@ -66,8 +66,35 @@ testaso! {
 }
 
 impl Secs {
-    /// TODO: The max size of an enclave should come from CPUID, which is inaccessible from within SGX.
-    pub const SIZE_MAX: u64 = 0x1_000_000_000;
+    /// Returns the maximum enclave size for 64bit in bytes.
+    /// CPUID.(EAX=12H, ECX=0H) enumerates Intel SGX capability;
+    /// For more on CPUID enumeration leaves, see 37.7.2 and Table 37-4.
```

Comment:
To document why this function is unsafe, add something like this:

```
/// # Safety
/// This function is unsafe because it does not check if the `CPUID` instruction is available before issuing it.
```

In the future, we can always make this function safe.

Reply:
Oh, you meant *literally* document! Sure thing.